### ISSUE TITLE
prefer-regex-literals should not suggest to use regex literal when the expression contains variable

### ISSUE THREAD

**Issue by Mogztter** [eslint/eslint#13843]
**What rule do you want to change?**

`prefer-regex-literals`

**Does this change cause the rule to produce more or fewer warnings?**

Arguably this is a bug and it should not suggest to use a regex literal because as far as I know it's simply not possible! So I guess fewer warnings :wink: 

**How will the change be implemented? (New option, new default behavior, etc.)?**

It should detect if the expression contains variables.
In the following examples, it should not suggest to use regex literal:

```
const regEx1 = new RegExp(`foo${var}`, 'gm')

const regEx2 = new RegExp('foo' + var, 'gm')
```

**Please provide some example code that this change will affect:**

See above :point_up: 

**What does the rule currently do for this code?**

It suggests to use a regex literal.

**What will the rule do after it's changed?**

Not suggest to use a regex literal :smile: 

**Are you willing to submit a pull request to implement this change?**

Why not, but I'm not familiar with the codebase so I will probably need some guidance :sweat_smile:

**Comment by Mogztter** [comment]
Sorry for the noise, it's late, I'm not looking at the right place! 😞
